### DISCUSSION TITLE
Add radon device class for sensors

### DISCUSSION THREAD

**Discussion by jrbeilke** [home-assistant/architecture#732] [Entity Models] [answered]
# Context

With the addition of the Airthings integration (https://www.home-assistant.io/integrations/airthings/) there are now devices and sensors tracking Radon levels (ie. Airthings View Plus), but Home Assistant does not currently offer a device class to handle units for these measurements.

For instance the Airthings API tracks Radon in becquerels (Bq/m³), but in the US picocuries (pCi/L) are more commonly used, and the current Home Assistant integration only supports the native unit of measurement (becquerels). This is confusing for Home Assistant users in the US (and elsewhere that are using Imperial units), as there is no easy way to change the units displayed for Radon levels of their devices.

Here are some links for additional context regarding measurement of Radon levels:
* https://help.airthings.com/en/articles/3119854-radon-what-is-the-difference-between-pci-l-and-bq-m3
* http://www.hec.uconn.edu/documents/MicrosoftWord-LM_04_What_are_units_formeasuring_9-26-06.pdf

# Proposal

Similar to what was done with the air quality units (ie https://github.com/home-assistant/architecture/discussions/573), I propose adding a new `radon` device class for sensors to Home Assistant, with support for unit conversion of becquerels and picocuries based on the imperial vs metric unit setting.

This would not only benefit the Airthings integration today, but also other air quality/monitoring integrations that include devices/sensors for tracking Radon levels as well.

Here is my pull request for adding the new radon device class: https://github.com/home-assistant/core/pull/66149

**Comment by MartinHjelmare** [comment]
Please try to list all the existing integrations in Home Assistant that could leverage this new device class and also potential new integrations known in the market that would benefit if integrated.

**Reply by jrbeilke** [reply]
Sure, a quick search in Gitlab shows that the Airthings integration is the only one that specifically has a `radon` SensorEntity currently in Home Assistant.

There is potential in the future for a Radon FTLab integration to support their RadonEye device, which is commonly used to measure Radon levels at home: http://radonftlab.com/radon-sensor-product/radon-detector/rd200/
(the RadonEye device measure levels in picocuries instead of becquerels like Airthings so unit conversion would be handy)
ie. https://community.home-assistant.io/t/radoneye-ble-interface/94962

This device class could also be used with sensors from the ESPHome integration in Home Assistant that are used to measure Radon, such as the Airthings Wave Plus and other devices that use Bluetooth ie.
https://github.com/esphome/feature-requests/issues/147

**Reply by jrbeilke** [reply]
Found one more `radon` sensor in Home Assistant custom-components:
https://github.com/custom-components/sensor.airthings_wave

**Reply by jrbeilke** [reply]
Almost forgot the new EcoQube Radon detector from Ecosense/FTLab. No Home Assistant integration yet that I'm aware of, but definitely seems to have potential:
https://ecosense.io/pages/ecoqube

**Reply by ch0gan** [reply]
Is this idea dead?  In addition to the need for radon unit conversion, 
![IMG_5857](https://github.com/home-assistant/architecture/assets/103752139/ca3d5433-c694-4fbc-a3e5-b8146c2b7b64)
![IMG_5856](https://github.com/home-assistant/architecture/assets/103752139/9ed65033-1b52-4d9f-8f4c-8f2cc0f88a34)
I also noticed pressure comes to HA as mbar but the imperial unit is hg.

**Reply by bdr99** [reply]
@ch0gan, the issue with the pressure unit for Airthings will be fixed in HA 2024.4. The fix was merged in https://github.com/home-assistant/core/pull/113005.

**Comment by bdraco** [comment]
picocuries and becquerel are both units of measure of radiation.  I think this makes more sense as a `radiation` sensor device class.

**Reply by jcgillespie** [reply]
picocuries and bequerels are both units of measure for radiation, but these devices aren't measuring strictly radiation. They measuring the radioactivity in a specific volume of air.

**Comment by jcgillespie** [comment]
I created a [PR](https://github.com/home-assistant/core/pull/134376) to implement this before realizing it would need a discussion in the architecture repo first.

**Comment by abmantis** [comment]
~~I am not sure if a device class should be this specific, since radon is just one gas out of many (despite being the one that most people will be affected by/deal with). Still, should the proposal be for something like `radiation concentration` instead?~~


EDIT:  We already have device classes that are not generic such as CO, CO², NO², so my previous argument does not make sense.

**Reply by jcgillespie** [reply]
I suggested that as well in #886 . I think the challenge will be balancing a larger change that is correct against a smaller immediately valuable change that may make future expansion harder.

**Reply by nethershaw** [reply]
I have one of the AirThings devices as well. Radon is a specifically regulated environmental contaminant here in the US, so the unit of measure (here, picocuries per liter; elsewhere, becquerels per cubic meter) are of specific interest to homeowners. People order so-called "radon reports," especially in areas like mine where soil uranium concentrations are relatively high.

Other radioactive airborne contaminants are typically not reported or measured at all in home environments because they are vastly less common outside of extremely specialist industrial situations; therefore, it _might not_ be especially useful to generalize from a sensor class of e.g. type `radon` to one of type `radiation concentration`, even though the units described by the former are a proper subset of the latter. Indeed the physical assumption made in consumer-grade radon sensors is that radon, being one of the few commonly and naturally occurring radioactive gases humans will encounter, is the only radioactive specimen present -- even though they technically can't distinguish between the decay events of one specimen from the decay events of another.

Possibly there are good reasons to imagine `radiation concentration` is useful for forward-compatibility, but the above constraints make that substantially less likely unless the target environment of Home Assistant is meant to include nuclear materials processing facilities, mines, and reactors. Just some context worth ~US$0.02.

**Reply by zar-ef** [reply]
I have a couple of air quality meters and dedicated Radon meters that both require Pi/L conversion template. Would have been nice to be able to select radon as a device_class and select what is unique unit of measurement. I second the opinion that scoping this to "radiation_concentration" implies measurements of other radiation gasses, which is highly unlikely outside of industrial facilities .

**Reply by abmantis** [reply]
So, asking the question the other way around: what is the downside of the `radiation concentration` device class approach?
As far as I know, the units (`Bq/m³` or `pCi/L`) would be the same with other gases.

**Reply by jcgillespie** [reply]
I can only think of one drawback - discoverability. Most people don’t associate these units with radiation concentration, just with radon. Their only meaningful exposure to these units is radon.

Given that the burden of discoverability is still not that great and primarily on integration developers, in my opinion that tradeoff is worthwhile given the alternative - technical debt. This is a shortcut that fits given current circumstances. If we accept the shortcut and circumstances change, it will be harder to adapt because of the things built atop the shortcut.

Given the conservative approach to architectural change, we should favor correctness that preserves options over most common use case unless it creates a bunch of complexity. In this case, the code is the same either way, we’re just talking about labeling.

**Reply by abmantis** [reply]
So, we do have a lot of device classes for individual gasses (CO, CO², NO², etc), in addition to the `GAS` device class (I wrongly assumed we they were mostly generic previously). Considering that then I am not against having a Radon specific device class.

**Reply by andorardo** [reply]
Does setting the device class to GAS limit the sort of units that are permitted? I tried to combine GAS with pCi/L, and it seemed to complain. (But I don't know how consequential that is.)

**Comment by mat1990dj** [comment]
Is there any progress, just came here to convert my radon measurements to pCi/L and found still no device class

**Comment by kbreit** [comment]
I'm interested in this as well and open to making a PR for it. But before I do I want to ensure there is some acceptance for implementing it.

**Reply by abmantis** [reply]
There is this related proposal too: https://github.com/home-assistant/architecture/discussions/886#discussioncomment-15641431

**Comment by abmantis** [comment]
To summarize, should we add a RADON device class with a UnitOfRadiationConcentration? The same unit could eventually be used for RADIATION_CONCENTRATION, but currently that would only benefit isy994, so it is not needed.

Current integrations that would benefit: 
- airthings
- airthings_ble
- airq
- aranet

**Comment by abmantis** [comment, marked as answer]
We've discussed this in the core architecture meeting and approved adding a `radon` device class along with `UnitOfRadiationConcentration`.

Thanks!

**Reply by abmantis** [reply]
Implemented in https://github.com/home-assistant/core/pull/175135
